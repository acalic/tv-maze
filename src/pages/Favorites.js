import React from "react";
import { useSelector } from "react-redux";
import { Alert } from "react-bootstrap";
import { FaInfoCircle } from "react-icons/fa";

import ShowsList from "@components/ShowsList";

const Favorites = () => {
  const showsFavorites = useSelector((state) => state.favorites.favorites);
  return (
    <>
      <h2>Favorites</h2>
      {showsFavorites && showsFavorites.length ? (
        <ShowsList shows={showsFavorites} />
      ) : (
        <Alert variant="light d-inline-flex align-items-center mb-5">
          <FaInfoCircle></FaInfoCircle>
          <span className="px-2">
            No favorites added yet, please click on the star icon next to the
            show title to add it to the list!
          </span>
        </Alert>
      )}
    </>
  );
};

export default Favorites;
