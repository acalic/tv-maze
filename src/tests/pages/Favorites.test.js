import React from "react";
import Favorites from "@pages/Favorites";
import { render, screen } from "@testing-library/react";

import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import { favorites } from "../fixtures";

describe("should test Favorites component", () => {
  const initialState = { favorites: favorites };
  const mockStore = configureStore();
  let store;

  it("renders the EpisodesList component", () => {
    store = mockStore(initialState);

    const wrapper = render(
      <Provider store={store}>
        <Favorites />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(screen.getByText(/Outer Banks/i)).toBeInTheDocument();
  });
});
