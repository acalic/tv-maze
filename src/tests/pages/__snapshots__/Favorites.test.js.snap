// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Favorites component renders the EpisodesList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        Favorites
      </h2>
      <div
        class="row"
      >
        <div
          class="py-3 d-flex col-lg-2 col-md-4 col-sm-6 col-6"
        >
          <div
            class="card-movie w-100 border-0 card"
          >
            <img
              class="card-img-top"
              src="https://static.tvmaze.com/uploads/images/medium_portrait/83/207960.jpg"
            />
            <div
              class="card-movie__body d-flex flex-column card-body"
            >
              <div
                class="d-flex justify-content-between align-items-center card-title h5"
              >
                Seinfeld
                 (
                1989
                )
                <button
                  class="btn-favorite btn btn-none btn-sm"
                  type="button"
                >
                  <svg
                    color="gold"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: gold;"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="text-muted card-text"
              >
                <span>
                  Comedy
                </span>
              </p>
              <p
                class="text-muted mt-auto card-text"
              >
                <span
                  class="mt-3"
                >
                  <span
                    class="text-secondary"
                  >
                    Rating:
                  </span>
                   
                  <span
                    class="text-success"
                  >
                    8.5
                  </span>
                </span>
              </p>
              <a
                class="btn btn-dark btn-md"
                href="/shows/530"
                role="button"
                tabindex="0"
              >
                See more
              </a>
            </div>
          </div>
        </div>
        <div
          class="py-3 d-flex col-lg-2 col-md-4 col-sm-6 col-6"
        >
          <div
            class="card-movie w-100 border-0 card"
          >
            <img
              class="card-img-top"
              src="https://static.tvmaze.com/uploads/images/medium_portrait/340/851075.jpg"
            />
            <div
              class="card-movie__body d-flex flex-column card-body"
            >
              <div
                class="d-flex justify-content-between align-items-center card-title h5"
              >
                Outer Banks
                 (
                2020
                )
                <button
                  class="btn-favorite btn btn-none btn-sm"
                  type="button"
                >
                  <svg
                    color="gold"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: gold;"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="text-muted card-text"
              >
                <span>
                  Adventure, Crime, Mystery
                </span>
              </p>
              <p
                class="text-muted mt-auto card-text"
              >
                <span
                  class="mt-3"
                >
                  <span
                    class="text-secondary"
                  >
                    Rating:
                  </span>
                   
                  <span
                    class="text-success"
                  >
                    6.8
                  </span>
                </span>
              </p>
              <a
                class="btn btn-dark btn-md"
                href="/shows/42062"
                role="button"
                tabindex="0"
              >
                See more
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2>
      Favorites
    </h2>
    <div
      class="row"
    >
      <div
        class="py-3 d-flex col-lg-2 col-md-4 col-sm-6 col-6"
      >
        <div
          class="card-movie w-100 border-0 card"
        >
          <img
            class="card-img-top"
            src="https://static.tvmaze.com/uploads/images/medium_portrait/83/207960.jpg"
          />
          <div
            class="card-movie__body d-flex flex-column card-body"
          >
            <div
              class="d-flex justify-content-between align-items-center card-title h5"
            >
              Seinfeld
               (
              1989
              )
              <button
                class="btn-favorite btn btn-none btn-sm"
                type="button"
              >
                <svg
                  color="gold"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: gold;"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
              </button>
            </div>
            <p
              class="text-muted card-text"
            >
              <span>
                Comedy
              </span>
            </p>
            <p
              class="text-muted mt-auto card-text"
            >
              <span
                class="mt-3"
              >
                <span
                  class="text-secondary"
                >
                  Rating:
                </span>
                 
                <span
                  class="text-success"
                >
                  8.5
                </span>
              </span>
            </p>
            <a
              class="btn btn-dark btn-md"
              href="/shows/530"
              role="button"
              tabindex="0"
            >
              See more
            </a>
          </div>
        </div>
      </div>
      <div
        class="py-3 d-flex col-lg-2 col-md-4 col-sm-6 col-6"
      >
        <div
          class="card-movie w-100 border-0 card"
        >
          <img
            class="card-img-top"
            src="https://static.tvmaze.com/uploads/images/medium_portrait/340/851075.jpg"
          />
          <div
            class="card-movie__body d-flex flex-column card-body"
          >
            <div
              class="d-flex justify-content-between align-items-center card-title h5"
            >
              Outer Banks
               (
              2020
              )
              <button
                class="btn-favorite btn btn-none btn-sm"
                type="button"
              >
                <svg
                  color="gold"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: gold;"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
              </button>
            </div>
            <p
              class="text-muted card-text"
            >
              <span>
                Adventure, Crime, Mystery
              </span>
            </p>
            <p
              class="text-muted mt-auto card-text"
            >
              <span
                class="mt-3"
              >
                <span
                  class="text-secondary"
                >
                  Rating:
                </span>
                 
                <span
                  class="text-success"
                >
                  6.8
                </span>
              </span>
            </p>
            <a
              class="btn btn-dark btn-md"
              href="/shows/42062"
              role="button"
              tabindex="0"
            >
              See more
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
